Make package

corinnebosley opened the pull request:
Make package importable by restructuring (a bit) and doing some other bits.

corinnebosley commented:
Might be best to review and merge license before merging this, just so that we can get everything in the right place, but I have tested the build using pip in a clean environment and it all seems to work.

crbunney commented on setup.py:
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+
+
+from setuptools import setup
+
+setup()
```
I don't believe this is required with recent versions of `setuptools` (although I don't know how recent that change is, nor do we have a formal stance on what versions we're committing to supporting)

crbunney reviewed (approved):
Looks comprehensive and good. I merged #9  if you wanted to retest with that in place

corinnebosley commented:
Thanks for merging #9 @crbunney , I will retest with that in place and hopefully get back to you shortly.

corinnebosley commented:
@crbunney I have added license and re-tested, all seems fine.  No license is in this PR, but I suppose that's just because you've already merged it.  I think this should all be good now.

@lucascatherine It would be good to get your go-ahead on this before I merge.  I have tested in a clean environment, all builds fine.

corinnebosley commented:
So it turns out that although the repo could be successfully packaged, it wasn't actually importable until these changes.
Import using 'import cap_sample_data'.

I may need to make some file structure changes in a further PR (for example, I am uneasy about shapefiles being on the top level, this may mean they are not importable or useable, and also it might be useful to add a symlink on the top level so that we can import using 'import cap_sample_data.sample_data', but right now I don't know how to do that.

I could do with this being merged (and imported into your respective cap_envs, should you use them) so that I can write tests that will pass.  That being the case (and with me not being around until next Monday), please merge ASAP.

corinnebosley commented:
As a further note to this PR, once this is merged I may have to change settings in ALL tests in cap, or maybe just this file (https://github.com/ADAQ-AQI/clean-air-project/blob/master/conftest.py) for the tests to pass again.

corinnebosley commented:
This is it guys, this one works and all the tests pass!!! I have moved all data files into cap_sample_data/sample_data and changed all the bits in the pyproject.toml and stuff to work properly, so now you can import the module using "import cap_sample_data" and get the data paths using "cap_sample_data.path" (this is what has changed in the conftest.py in my next PR to cap).

I am happy with this; ready to merge upon approval from @lucascatherine .

lucascatherine commented:
Looks good to me.

lucascatherine reviewed (approved):
All looks good.
